<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>M_hSmDialog</class>
 <widget class="QDialog" name="M_hSmDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>720</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MH ↔ SM Plugin</string>
  </property>

  <layout class="QVBoxLayout" name="verticalLayout_main">
   <property name="spacing">
    <number>10</number>
   </property>

   <!-- ===================================================== -->
   <!-- PIPELINE 1 : MH -> SM -->
   <!-- ===================================================== -->
   <item>
    <widget class="QGroupBox" name="group_export">
     <property name="title">
      <string>Pipeline MH vers SM</string>
     </property>
     <layout class="QGridLayout" name="grid_export">
      <property name="horizontalSpacing">
       <number>8</number>
      </property>
      <property name="verticalSpacing">
       <number>8</number>
      </property>

      <!-- Colonnes : label / champ / bouton -->
      <item row="0" column="0">
       <widget class="QLabel" name="label_mh">
        <property name="text">
         <string>Entrée MH</string>
        </property>
        <property name="toolTip">
         <string>Raster MH au format ESRI ASCII grid (.txt ou .asc)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="line_mh">
        <property name="placeholderText">
         <string>Choisir un fichier MH (.txt ou .asc)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="btn_mh">
        <property name="text">
         <string>Parcourir</string>
        </property>
        <property name="minimumWidth">
         <number>110</number>
        </property>
       </widget>
      </item>

      <item row="1" column="0">
       <widget class="QLabel" name="label_cir">
        <property name="text">
         <string>Fichier CIR</string>
        </property>
        <property name="toolTip">
         <string>scene_triangle.cir (ordre officiel des triangles)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="line_cir">
        <property name="placeholderText">
         <string>Choisir le fichier .cir associé au raster </string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="btn_cir">
        <property name="text">
         <string>Parcourir</string>
        </property>
        <property name="minimumWidth">
         <number>110</number>
        </property>
       </widget>
      </item>

      <item row="2" column="0">
       <widget class="QLabel" name="label_val_out">
        <property name="text">
         <string>Sortie SM</string>
        </property>
        <property name="toolTip">
         <string>Fichier .val généré (une valeur par triangle, facette par facette)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="line_val">
        <property name="placeholderText">
         <string>Choisir le fichier .val de sortie</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="btn_val">
        <property name="text">
         <string>Choisir</string>
        </property>
        <property name="minimumWidth">
         <number>110</number>
        </property>
       </widget>
      </item>

      <item row="3" column="0" colspan="3">
       <widget class="QPushButton" name="btn_export">
        <property name="minimumHeight">
         <number>38</number>
        </property>
        <property name="text">
         <string>Lancer MH vers SM</string>
        </property>
       </widget>
      </item>

      <!-- Ajustement des largeurs -->
      <item row="4" column="0">
       <spacer name="spacer_export_left">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
       </spacer>
      </item>

     </layout>
    </widget>
   </item>

   <!-- ===================================================== -->
   <!-- PIPELINE 2 : SM -> MH -->
   <!-- ===================================================== -->
   <item>
    <widget class="QGroupBox" name="group_inverse">
     <property name="title">
      <string>Pipeline SM vers MH</string>
     </property>
     <layout class="QGridLayout" name="grid_inverse">
      <property name="horizontalSpacing">
       <number>8</number>
      </property>
      <property name="verticalSpacing">
       <number>8</number>
      </property>

      <item row="0" column="0">
       <widget class="QLabel" name="label_val_in">
        <property name="text">
         <string>Entrée SM</string>
        </property>
        <property name="toolTip">
         <string>Fichier .val existant à reconvertir en raster MH</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="line_val_in">
        <property name="placeholderText">
         <string>Choisir un fichier SM (.val)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="btn_val_in">
        <property name="text">
         <string>Parcourir</string>
        </property>
        <property name="minimumWidth">
         <number>110</number>
        </property>
       </widget>
      </item>

      <item row="1" column="0" colspan="3">
       <widget class="QLabel" name="label_sortie_auto">
        <property name="text">
         <string>Sortie automatique : le raster sera écrit dans le même dossier que le fichier .val, sous la forme nom2asc.asc</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="styleSheet">
         <string>color: #555; padding: 6px; background: #f5f5f5; border: 1px solid #ddd; border-radius: 4px;</string>
        </property>
       </widget>
      </item>

      <item row="2" column="0" colspan="3">
       <widget class="QPushButton" name="btn_reconstruire">
        <property name="minimumHeight">
         <number>38</number>
        </property>
        <property name="text">
         <string>Lancer Pipeline SM vers MH</string>
        </property>
       </widget>
      </item>

      <item row="3" column="0" colspan="3">
       <widget class="QPushButton" name="btn_comparer">
        <property name="minimumHeight">
         <number>38</number>
        </property>
        <property name="text">
         <string>Comparer MH et raster reconstruit</string>
        </property>
       </widget>
      </item>

     </layout>
    </widget>
   </item>

   <!-- ===================================================== -->
   <!-- OUTILS -->
   <!-- ===================================================== -->
   <item>
    <widget class="QGroupBox" name="group_tools">
     <property name="title">
      <string>Outils</string>
     </property>
     <layout class="QHBoxLayout" name="hbox_tools">
      <property name="spacing">
       <number>10</number>
      </property>
      <item>
       <widget class="QPushButton" name="btn_visu_mh">
        <property name="text">
         <string>Afficher le raster MH dans QGIS</string>
        </property>
        <property name="minimumHeight">
         <number>30</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="spacer_tools_right">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>

   <!-- ===================================================== -->
   <!-- MED (OPTIONNEL) -->
   <!-- ===================================================== -->
   <item>
    <widget class="QGroupBox" name="group_med">
     <property name="title">
      <string>MED optionnel</string>
     </property>
     <layout class="QGridLayout" name="grid_med">
      <property name="horizontalSpacing">
       <number>8</number>
      </property>
      <property name="verticalSpacing">
       <number>8</number>
      </property>

      <item row="0" column="0">
       <widget class="QLabel" name="label_med">
        <property name="text">
         <string>Fichier MED</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="line_med">
        <property name="placeholderText">
         <string>Choisir un fichier MED (.med)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="btn_med">
        <property name="text">
         <string>Parcourir</string>
        </property>
        <property name="minimumWidth">
         <number>110</number>
        </property>
       </widget>
      </item>

      <item row="1" column="0" colspan="3">
       <widget class="QPushButton" name="btn_visu_med">
        <property name="minimumHeight">
         <number>30</number>
        </property>
        <property name="text">
         <string>Visualiser MED avec PyVista</string>
        </property>
       </widget>
      </item>

     </layout>
    </widget>
   </item>

   <item>
    <spacer name="spacer_bottom">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>

  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>

